internal/services/monitor/migration/diagnostic_setting_test.go

```diff
@@ -0,0 +1,63 @@
+package migration
+
+import (
+	"context"
+	"testing"
+
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func TestDiagnosticSettingV0ToV1(t *testing.T) {
+	testData := []struct {
+		name     string
+		input    map[string]interface{}
+		expected *string
+	}{
+		{
+			name: "old id",
+			input: map[string]interface{}{
+				"id": "/subscriptions/12345678-1234-5678-1234-123456789012/resourcegroups/group1/providers/Microsoft.KeyVault/vaults/vault1|logMonitoring1",
```

Comment:
Is it possible for the ID has no leading slash, like `subscriptions/12345678-1234-5678-1234-123456789012/resourcegroups/group1/providers/Microsoft.KeyVault/vaults/vault1|logMonitoring1`

Reply:
how would we get an ID that starts without a /? that seems like a bug and we should report int, link where it happens, and then just pop a / on the front before parse?

same goes with the incorrect casing, on read we just correct it and pop it into the parse function with a link to the bug. everything coming from the config side should be correct?

Reply:
Yes that everything from the config side should be correct, and I will add a validate in the `ParseMonitorDiagnosticId` func to filter coming IDs.